src/menu/index.tsx

```diff
@@ -0,0 +1,314 @@
+		if (onActiveIndexChange) {
+			onActiveIndexChange(index);
+		} else {
+			icache.set('activeIndex', index);
+		}
+	}
+
+	function setValue(value: string) {
+		icache.set('value', value);
+		onValue(value);
+	}
+
+	function onKeyDown(event: KeyboardEvent) {
+		event.stopPropagation();
+
+		switch (event.which) {
+			case Keys.Enter:
+			case Keys.Space:
+				event.preventDefault();
+				const activeItem = options[computedActiveIndex];
+				!activeItem.disabled && setValue(activeItem.value);
+				break;
+			case Keys.Down:
+				event.preventDefault();
+				setActiveIndex((computedActiveIndex + 1) % options.length);
+				break;
+			case Keys.Up:
+				event.preventDefault();
+				setActiveIndex((computedActiveIndex - 1 + options.length) % options.length);
+				break;
+			case Keys.Escape:
+				event.preventDefault();
+				onRequestClose && onRequestClose();
+				break;
+			case Keys.Home:
+				event.preventDefault();
+				setActiveIndex(0);
+				break;
+			case Keys.End:
+				event.preventDefault();
+				setActiveIndex(options.length - 1);
+				break;
+			default:
+				const newIndex = getComputedIndexFromInput(event.key);
+				if (newIndex !== undefined) {
+					setActiveIndex(newIndex);
+				}
+		}
+	}
+
+	function getComputedIndexFromInput(key: string) {
+		const existingTimer = icache.get('resetInputTextTimer');
+		let inputText = icache.getOrSet('inputText', '') + `${key}`;
+		existingTimer && clearTimeout(existingTimer);
+
+		const resetTextTimeout = setTimeout(() => {
+			icache.set('inputText', '');
+		}, 800);
+
+		icache.set('resetInputTextTimer', resetTextTimeout);
+		icache.set('inputText', inputText);
+
+		return findIndex(
+			options,
+			({ disabled, value, label }) =>
+				!disabled && (label || value).toLowerCase().indexOf(inputText.toLowerCase()) === 0
+		);
+	}
+
+	function onActive(index: number, itemDimensions: DimensionResults) {
+		const { position: itemPosition, size: itemSize } = itemDimensions;
+		const { position: rootPosition, size: rootSize } = dimensions.get('root');
+		if (itemPosition.bottom > rootPosition.bottom) {
+			const numInView = Math.ceil(rootSize.height / itemSize.height);
+			icache.set('itemToScroll', Math.max(index - numInView + 1, 0));
+		} else if (itemPosition.top < rootPosition.top) {
+			icache.set('itemToScroll', index);
+		}
+	}
+
+	const itemToScroll = icache.get('itemToScroll');
+	const menuHeight = icache.get('menuHeight');
+	const idBase = icache.getOrSet('idBase', uuid());
+	const rootStyles = menuHeight ? { maxHeight: `${menuHeight}px` } : {};
+	const shouldFocus = focus.shouldFocus();
+	const classes = theme.classes(css);
+
+	return (
+		<div
+			key="root"
+			classes={classes.menu}
+			tabIndex={focusable ? 0 : -1}
+			onkeydown={onKeyDown}
+			focus={() => shouldFocus}
+			onfocus={onFocus}
+			onblur={onBlur}
+			styles={rootStyles}
+			role={listBox ? 'listbox' : 'menu'}
+			aria-orientation="vertical"
+			aria-activedescendant={`${idBase}-item-${activeIndex}`}
```

Comment:
This should not be set if there if `activeIndex == null`. For example, on the deployed site for this PR, the example menu's `aria-activedescendant` is always `{id}-undefined` since the active index is never set.

Reply:
fixing that now